### PR TITLE
DAOS-7029 control: Add refresh methods for agent cache

### PR DESCRIPTION
**Pull request by kjacque**
- Add agent config option for a cache refresh interval. The agent will perform a lazy cache refresh when the configured interval elapses.
- Allow users to force-refresh the caches with SIGUSR2. 

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/control/lib/cache/cache.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
+//
+// (C) Copyright 2023 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package cache
```
Comment (tanabarr) [inline src/control/lib/cache/cache.go:7]: is https://pkg.go.dev/istio.io/pkg/cache unsuitable? do we need to roll our own?

**Review by tanabarr — APPROVED** [review]
don't see any problems with the PR, only question is whether we could use an existing (mature) cache library rather than rolling our own?

File: src/control/lib/cache/cache.go (hunk 1 above)
Comment (kjacque) [reply src/control/lib/cache/cache.go:7]: The cache lib ends up being pretty simple, especially with some changes suggested by Mike. I think rolling our own makes sense here, as the functionality in the linked package is also a little different from what we're looking for.

File: src/control/lib/cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+	ic.mutex.RLock()
+	defer ic.mutex.RUnlock()
+
+	item, err := ic.get(key)
+	if err != nil {
+		return nil, noopRelease, err
+	}
+
+	if item.NeedsRefresh() {
+		if err := item.Refresh(ctx); err != nil {
```
Comment (mjmac) [inline src/control/lib/cache/cache.go:163]: My only hesitation with this pattern (and the reason I removed `NeedsRefresh()` from the interface) is that it's potentially racy. The `NeedsRefresh()` check and `Refresh()` actions are done under different locks. It's not unsafe, as far as I can tell, but could result in a refresh being done unnecessarily if more than one goroutine does the `NeedsRefresh()` check nearly simultaneously.

Comment (kjacque) [reply src/control/lib/cache/cache.go:163]: Good point, I'll revisit this.

File: src/control/cmd/daos_agent/config_test.go [hunk 3]
```diff
@@ -175,3 +178,28 @@ transport_config:
 		})
 	}
 }
+
+/*func TestAgent_Config_CacheRefreshInterval(t *testing.T) {
```
Comment (tanabarr) [inline src/control/cmd/daos_agent/config_test.go:182]: should this be removed or uncommented?

**Review by tanabarr — DISMISSED** [review]

File: src/control/cmd/daos_agent/config_test.go (hunk 3 above)
Comment (kjacque) [reply src/control/cmd/daos_agent/config_test.go:182]: Ah, good catch, I forgot to remove it since it was no longer necessary.

**Review by mjmac — APPROVED** [review]

**Pull request merged**